src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -1009,6 +1009,53 @@ public function testMirrorCopiesRelativeLinkedContents()
         $this->assertEquals('\\' === DIRECTORY_SEPARATOR ? realpath($sourcePath.'\nested') : 'nested', readlink($targetPath.DIRECTORY_SEPARATOR.'link1'));
     }
 
+    public function testMirrorContentsWithSameNameAsSourceOrTargetWithoutDeleteOption()
+    {
+        $sourcePath = $this->workspace.DIRECTORY_SEPARATOR.'source'.DIRECTORY_SEPARATOR;
+
+        $oldPath = getcwd();
```

Comment:
Doing clean up work after performing assertions is not reliable. As soon as one of the assertions fails clean up work will not happen anymore. So what we could is remember the current directory inside the base class' `setUp()` method and restore it in `tearDown()`.